cmd/containerd-shim-runhcs-v1/task.go

```diff
@@ -80,6 +80,10 @@ type shimTask interface {
 	//
 	// If the host is not hypervisor isolated returns `""`.
 	DumpGuestStacks(ctx context.Context) string
+	// Share shares a directory/file into the host UVM.
+	//
+	// If the host is not hypervisor isolated returns error.
```

Comment:
Does it not work to do a MappedDirectory modify request for a wcow-process container?

Reply:
No I'm sure it does, all of the shimdiag things mostly relate to the UVM is why. Would you just wanna provide a container ID as the last parameter and if it's omitted just share to the shims UVM? For instance

./shimdiag share k8s.io-<id> /path/on/host /path/in/container/ <containerid> 

May be missing something easier but if this was going to be shared into a process isolated container that's in a pod we'd have to call service.getPod() ---> pod.GetTask(id) ---> see if it's an hcsTask ---> type assert task.c to hcs.System and then call Modify?

Reply:
I think what you have now is okay, just checking if we expect container modify to work in case we want to do that in the future.